tmt/steps/report/reportportal.py

```diff
@@ -0,0 +1,89 @@
+import dataclasses
+import io
+import os
+import zipfile
+from typing import List, Optional
+
+import click
+import junit_xml
+
+import tmt.steps.report
+import tmt.utils
+
+from . import junit
+
+
+@dataclasses.dataclass
+class ReportPortalData(tmt.steps.StepData):
+    url: Optional[str] = None
+    token: Optional[str] = None
+    launch_name: Optional[str] = None
+
+
+@tmt.steps.provides_method("reportportal")
+class ReportPortal(tmt.steps.report.ReportPlugin):
+    """
+    Send results in JUnit format to Report Portal instance.
+    """
+
+    _data_class = ReportPortalData
+
+    @classmethod
+    def options(cls, how: Optional[str] = None) -> List[tmt.options.ClickOptionDecoratorType]:
+        """ Prepare command line options """
+        options = super().options(how)
+        options[:0] = [
+            click.option(
+                "--url", envvar="TMT_REPORT_REPORTPORTAL_URL", metavar="URL", required=True,
+                help="The URL of Report Portal instance where the data should be sent to"),
+            click.option(
+                "--token", envvar="TMT_REPORT_REPORTPORTAL_TOKEN", metavar="TOKEN", required=True,
+                help="The token to use for upload to Report Portal instance"),
+            click.option(
+                "--launch_name", metavar="LAUNCH_NAME",
```

Comment:
Probably `--launch-name` would be better, more fitting to using `-` as a separator of option name components. Also, `NAME` as a metavar would be perfectly fine, `--launch-name=NAME` is fine (my two cents...).

Reply:
Ok, addressed in the latest commit.